Homework-4/Operations/DictionaryOpeartions.cs

```diff
@@ -0,0 +1,163 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Homework_4.Operations
+{
+    internal class DictionaryOpeartions
+    {
+        private Dictionary<int, double> dictionaryItems = new Dictionary<int, double>();
+        
+        /// <summary>
+        /// Executes dictionary operations.
+        /// </summary>
+        public void Operations() {
+            ListOperations listOperations = new ListOperations();
+            while (true)
+            {
+                listOperations.PrintMenu("Dictionary");
+                Console.Write("Enter your choice : ");
+                if (!int.TryParse(Console.ReadLine(), out int choice))
+                {
+                    Console.WriteLine("Invalid input. Please enter a valid number.");
+                    continue;
+                }
+                switch(choice)
+                {
+                    case 1:
+                        AddElement();
+                        break;
+                    case 2:
+                        PrintElements();
+                        break;
+                    case 3:
+                        DeleteElement(0,"First");
+                        break;
+                    case 4:
+                        int middleIndex=(dictionaryItems.Count%2==0)? dictionaryItems.Count / 2-1: dictionaryItems.Count / 2;
+                        DeleteElement(middleIndex, "Middle");
+                        break;
+                    case 5:
+                        DeleteElement(dictionaryItems.Count-1, "Last");
+                        break;
+                    case 6:
+                        CalculateAverage();
+                        break;
+                    case 7:
+                        return;
+                        break;
+                    case 8:
+                        Environment.Exit(0);
+                        break;
+                    default:
+                        Console.WriteLine("Invalid input. Please enter a valid number.");
+                        break;
+                }
+            }
+        }
+        /// <summary>
+        /// Adds an element to the dictionary.
+        /// </summary>
+        public void AddElement()
```

Comment:
Interface should have defined the methods and both dict and list classes should have implemented them separately.